### ISSUE TITLE
the processed data of mimiciii dataset does not match the train_listfile

### ISSUE THREAD

**Issue by liugaohong** [choczhang/ConCare#4]
when I get the data to train, I found that the  data is different from the train_listfile
so I reclassify the data to get the new train_data、valid_data and test_data, and renormalize the train_data and get the new normalize weight ihm_normalize_new
but when I train the model, the auc_of_roc droped by 3%, almost 84.6%.

not all，when I run the model, I find that when calculating the decorrelation_loss, it can use matrix to calculation directly, but in the code is two loop used. 
also, the result of the new code after the modification was reduced to about 81%

Is there a problem with my data processing? Or beacuse the effect of calculating the de_correlation_loss  is a matrix way is not very good?

**Comment by choczhang** [comment]
Our proposed ConCare model has been developed on the previous version of the MIMIC-III and the corresponding benchmark (Ver. 2018). Implementation details of the normalizer and the dataset partitioning have been changed by the original publisher of the dataset and the benchmark. To help researchers reproduce the model, we have released some of the previous files below in google drive:

train val : https://drive.google.com/file/d/1066eTdXhZczTh1Nf4WTA2l1lv5-6S4pI/view?usp=sharing

test: https://drive.google.com/file/d/1KHRPLznKMFi4s1hCxDxAjGWBPPgHXxj7/view?usp=sharing

demographic: https://drive.google.com/file/d/1TXn4UdtQCzfd7TdDJAo_6_IcnO2LUa1a/view?usp=sharing

**Comment by liugaohong** [comment]
thanks, there have another questtion! in the ConCare, the decorrelation_loss  is calculated as follows:
```python
#DeCov
DeCov_contexts = x.transpose(0, 1).transpose(1, 2) # I+1 H B
Covs = cov(DeCov_contexts[0,:,:])
DeCov_loss = 0.5 * (torch.norm(Covs, p = 'fro')**2 - torch.norm(torch.diag(Covs))**2 ) 
for i in range(feature_dim -1 + 1):
    Covs = cov(DeCov_contexts[i+1,:,:])
    DeCov_loss += 0.5 * (torch.norm(Covs, p = 'fro')**2 - torch.norm(torch.diag(Covs))**2 )
```
```python
def cov(m, y=None): #m:64*256
    if y is not None:
        m = torch.cat((m, y), dim=0)
    m_exp = torch.mean(m, dim=1)
    x = m - m_exp[:, None]
    cov = 1 / (x.size(1) - 1) * x.mm(x.t()) # matrix multiplication
    return cov
```

this method does not operate on matrix, and it's not meet the  C<sub>i,j</sub> and L<sub>decorrelation</sub>
there is my code as peer the paper

```python
#DeCov 
cs = cross_matrix(x)
DeCor_loss = 0.5 * ((cs**2).sum() - ((torch.diag(cs))**2).sum() )
x = self.final_linear(x)
return x, DeCor_loss
```
```python
def cross_matrix(x):
    x_mean = torch.mean(x, 0, True)
    x_mean = x_mean.expand(x.shape[0], -1, -1)
    x = torch.sub(x, x_mean)
    x = torch.matmul(x, (x.transpose(1,2)))
    x = 1 / (x.size(1)) * torch.sum(x, 0)

    return x
```

but the auc_of_roc result of my code is only about 81%(accordling to my own classified data and my own normalize)
So I want to ask is my approach incorrect? Why do you directly use a feature to calculate the correlation loss between this feature and itself, and then do the accumulation, instead of using a matrix to directly calculate the correlation loss between each feature.
